orca-queue/src/test/kotlin/com/netflix/spinnaker/orca/q/handler/RunTaskHandlerTest.kt

```diff
@@ -1839,6 +1831,70 @@ object RunTaskHandlerTest : SubjectSpek<RunTaskHandler>({
       }
     }
   }
+
+  describe("when a stage had skipped manual judgment stage with auth propagated") {
```

Comment:
Would this be clearer as:
```suggestion
  describe("when a previous stage was a skipped manual judgment stage with auth propagated") {
```

Reply:
Plus also, I ran these tests locally with this diff:
```
diff --git a/orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/AuthenticationAware.kt b/orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/AuthenticationAware.kt
index 59c16b120..6e4ea8df2 100644
--- a/orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/AuthenticationAware.kt
+++ b/orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/AuthenticationAware.kt
@@ -56,7 +56,8 @@ interface AuthenticationAware {
     return stageNavigator
       .ancestors(stage)
       .firstOrNull { it.stageBuilder is AuthenticatedStage } ?.let{
-        (it.stageBuilder as AuthenticatedStage).authenticatedUser(solveSkippedStages(it.stage)).orElse(null)
+      //        (it.stageBuilder as AuthenticatedStage).authenticatedUser(solveSkippedStages(it.stage)).orElse(null)
+        (it.stageBuilder as AuthenticatedStage).authenticatedUser(it.stage).orElse(null)
       }
   }
 ```
and I expected them to fail, but they still pass.

Reply:
Solved, try run again, I was not mocking the ancestor call to return something, so always was returning null, but now is checking working as expected

Reply:
It does fail now, thank you.  The failure is:
```
org.mockito.exceptions.verification.TooFewActualInvocations: 
stageNavigator.ancestors(
    Stage {id='01GBRAFJC4SSD9AC74E5EZBAV3', executionId='01GBRAFJAFQF2443HFYHJSTN1P'}
);
Wanted 2 times:
-> at com.netflix.spinnaker.orca.pipeline.util.StageNavigator.ancestors(StageNavigator.java:46)
But was 1 time:
-> at com.netflix.spinnaker.orca.q.handler.AuthenticationAware$DefaultImpls.retrieveAuthenticatedUser(AuthenticationAware.kt:57)

	at com.netflix.spinnaker.orca.pipeline.util.StageNavigator.ancestors(StageNavigator.java:46)
	at com.netflix.spinnaker.orca.q.handler.RunTaskHandlerTest$1$8$1$5.invoke(RunTaskHandlerTest.kt:1923)
	at com.netflix.spinnaker.orca.q.handler.RunTaskHandlerTest$1$8$1$5.invoke(RunTaskHandlerTest.kt:86)
```
from this assertion:
```
      it("verify stage backtracking times") {
        verify(stageNavigator, times(2)).ancestors(stage)
      }
```
Enough time has passed that I don't remember the details of the PR well enough.  Is there a way to structure the test so there's a more "direct" failure -- like a pipeline fails where it would otherwise succeed, or some kind of auth failure, or something?  Counting the number of times a method gets called seems a little fragile.

Reply:
The gap here is the task context, because I am not able to send a ManualJudgment stage SKIPPED that will crash by authenticated issues, it never crash because by default will take the execution user (that is the current bug), so go into the ancestors more that 1 time means that a backtracking was applied (By the moment I changed for 'atLeast' for remove some fragility in the test)

Reply:
Are you on spinnaker slack?  I'd love to chat about this.

Reply:
I added my account to spinnaker slack now, let me send you a message :D

Reply:
Hey @dbyron-sf from our yesterday talk, I applied some changes to test directly the class ManualJudgmentStage, and I cleaned the code to have a direct test case. Again, tank you to talked with me, your advices were very helpfull

Reply:
Excellent!